source/developers-guide/example-plugin/index.md

```diff
@@ -28,187 +26,201 @@ This example gives answers to the following questions:
 
-        $this->subscribeEvent(
-            'Enlight_Bootstrap_AfterInitResource_shopware_storefront.list_product_service',
-            'registerListProductService'
-        );
+```php
+<?php
 
-        $this->subscribeEvent(
-            'Enlight_Controller_Action_PostDispatchSecure_Frontend',
-            'addTemplateDir'
-        );
+namespace SwagPluginSystem\Subscriber;
 
-        $this->subscribeEvent(
-            'Enlight_Controller_Action_PostDispatchSecure_Widgets',
-            'addTemplateDir'
-        );
+use Enlight\Event\SubscriberInterface;
 
-        $this->subscribeEvent(
-            'Theme_Compiler_Collect_Plugin_Less',
-            'onCollectLessFiles'
-        );
+class TemplateRegistration implements SubscriberInterface
+{
+    /**
+     * @var string
+     */
+    private $pluginDirectory;
 
-        return true;
-    }
+    /**
+     * @var \Enlight_Template_Manager
+     */
+    private $templateManager;
 
-    public function afterInit()
+    /**
+     * @param $pluginDirectory
+     * @param \Enlight_Template_Manager $templateManager
+     */
+    public function __construct($pluginDirectory, \Enlight_Template_Manager $templateManager)
     {
-        $this->get('Loader')->registerNamespace(
-            'ShopwarePlugins\SwagPluginSystem',
-            $this->Path()
-        );
+        $this->pluginDirectory = $pluginDirectory;
+        $this->templateManager = $templateManager;
     }
 
     /**
-     * @return \Shopware\Components\Theme\LessDefinition
+     * {@inheritdoc}
      */
-    public function onCollectLessFiles()
+    public static function getSubscribedEvents()
     {
-        return new \Shopware\Components\Theme\LessDefinition(
-            [],
-            [__DIR__ . '/Views/frontend/_public/src/less/all.less']
-        );
+        return [
+            'Enlight_Controller_Action_PreDispatch' => 'onPreDispatch'
+        ];
     }
 
-    public function addTemplateDir()
+    public function onPreDispatch()
     {
-        Shopware()->Container()->get('template')->addTemplateDir($this->Path() . 'Views/');
+        $this->templateManager->addTemplateDir($this->pluginDirectory . '/Resources/views');
     }
+}
+```
 
-    public function registerSeoCategoryService()
-    {
-        $seoCategoryService = new SeoCategoryService(
-            Shopware()->Container()->get('dbal_connection'),
-            Shopware()->Container()->get('shopware_storefront.category_service')
-        );
-        Shopware()->Container()->set('shopware_storefront.seo_category_service', $seoCategoryService);
-    }
+The handler `onPreDispatch` registers the plugin's `/Resources/views` directory as a template directory for Shopware.
+Attention: This event listener listens to the global pre dispatch event. The plugin shouldn't do some performance sensitive tasks here, otherwise each post dispatch event in the store front will be slowed down.
```

Comment:
, otherwise each! request will be slowed down

Reply:
f